pallet/staking/src/lib.rs

```diff
@@ -181,18 +211,62 @@ pub mod pallet {
 	#[pallet::getter(fn collator_of)]
 	pub type Collators<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, Perbill>;
 
-	/// Current stakers' exposure.
+	/// Exposure cache states.
+	///
+	/// To avoid extra DB RWs during new session, such as:
```

Comment:
Honestly, this is indeed a great solution to avoid copying the current session's callator information into another container. However, according to my understanding of this problem, ExposureCacheState is likely a pointer to either the current session storage in the elect() or the previous session storage in the hook, it switches as the session switches. So, I think two pointers are enough, now there are three.

Reply:
How to implement this with two pointers?